src/test/java/com/epam/tc/hw3/pages/BaseJdiTestingPage.java

```diff
@@ -0,0 +1,123 @@
+package com.epam.tc.hw3.pages;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class BaseJdiTestingPage {
+
+    protected final WebDriver driver;
+    protected final WebDriverWait wait;
+    protected final SoftAssertions softAssertions;
```

Comment:
Page Object is not proper place for the assertions. All assertions should be invoke on a test level or step level